### PR TITLE
Benchmark regression tests

### PR DESCRIPTION
**Pull request by kanaadp**
Modified benchmark run scripts to be more general, added script to run all benchmarks. We still need a script to pull & evaluate trained benchmarks.

### CODE REVIEW

File: flow/benchmarks/rllib/es_runner.py [hunk 1]
```diff
@@ -33,11 +79,15 @@
...
     config = agent_cls._default_config.copy()
-    config["episodes_per_batch"] = N_ROLLOUTS
-    config["num_workers"] = N_ROLLOUTS
+    config["num_workers"] = min(num_cpus, num_rollouts)
+    config["episodes_per_batch"] = num_rollouts
     config["eval_prob"] = 0.05
     config["noise_stdev"] = grid_search([0.01, 0.02])
     config["stepsize"] = grid_search([0.01, 0.02])
+
+    config["noise_stdev"] = 0.02
```
Comment (AboudyKreidieh) [inline flow/benchmarks/rllib/es_runner.py]: we should set the hp based on the best performing for each benchmark (see paper for details)

File: flow/benchmarks/rllib/es_runner.py [hunk 2]
```diff
@@ -60,6 +110,6 @@
             "max_failures": 999,
             "stop": {"training_iteration": 500},
             "num_samples": 1,
-            "upload_dir": "s3://<BUCKET NAME>"
+            "upload_dir": "s3://" + upload_dir
```
Comment (AboudyKreidieh) [inline flow/benchmarks/rllib/es_runner.py]: what if you're not running experiments on aws? this should be defaulted to no uploading there

File: flow/benchmarks/rllib/ppo_runner.py [hunk 3]
```diff
@@ -68,6 +112,6 @@
                 "training_iteration": 500
             },
             "num_samples": 3,
-            "upload_dir": "s3://<BUCKET NAME>"
+            "upload_dir": "s3://" + upload_dir
```
Comment (AboudyKreidieh) [inline flow/benchmarks/rllib/ppo_runner.py]: again

File: flow/benchmarks/rllib/ppo_runner.py [hunk 4]
```diff
@@ -35,12 +79,12 @@
...
-    config["num_workers"] = min(N_CPUS, N_ROLLOUTS)
-    config["train_batch_size"] = horizon * N_ROLLOUTS
+    config["num_workers"] = min(num_cpus, num_rollouts)
+    config["train_batch_size"] = horizon * num_rollouts
     config["use_gae"] = True
     config["horizon"] = horizon
-    config["lambda"] = grid_search([0.97, 1.0])
-    config["lr"] = grid_search([5e-4, 5e-5])
+    config["lambda"] = 0.97
+    config["lr"] = 5e-4
```
Comment (AboudyKreidieh) [inline flow/benchmarks/rllib/ppo_runner.py]: again

File: flow/benchmarks/run_all_benchmarks.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
```
Comment (AboudyKreidieh) [inline flow/benchmarks/run_all_benchmarks.sh]: nice! can we document this somewhere though? (either a the start of this script or in the benchmarks' `README.md`)

File: scripts/benchmark_autoscale.yaml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+# For more documentation on available fields, see:
+# http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.ServiceResource.create_instances
+head_node:
+    InstanceType: c4.4xlarge
+    ImageId: ami-060b3311dd2b675ee # Flow AMI (Ubuntu)
+    InstanceMarketOptions:
+        MarketType: spot
+         #Additional options can be found in the boto docs, e.g.
+#        SpotOptions:
+#            MaxPrice: "1.0"
```
Comment (AboudyKreidieh) [inline scripts/benchmark_autoscale.yaml]: nit: should prob keep this on

File: flow/benchmarks/rllib/es_runner.py [hunk 7]
```diff
@@ -33,11 +79,15 @@
 
     agent_cls = get_agent_class(alg_run)
     config = agent_cls._default_config.copy()
-    config["episodes_per_batch"] = N_ROLLOUTS
-    config["num_workers"] = N_ROLLOUTS
+    config["num_workers"] = min(num_cpus, num_rollouts)
+    config["episodes_per_batch"] = num_rollouts
     config["eval_prob"] = 0.05
     config["noise_stdev"] = grid_search([0.01, 0.02])
```
Comment (eugenevinitsky) [inline flow/benchmarks/rllib/es_runner.py]: specifically, we can remove these duplicate lines?

**Comment by eugenevinitsky** [comment]
@kanaadp can we move this to the scripts folder and add some documentation to a README?

**Comment by eugenevinitsky** [comment]
@kanaadp tests are failing here so we can't merge it

File: flow/benchmarks/run_all_benchmarks.sh (hunk 5 above)
Comment (kanaadp) [reply flow/benchmarks/run_all_benchmarks.sh]: peep the README.md

**Comment by eugenevinitsky** [comment]
@kanaadp all comments seem addressed; can we merge?

**Pull request merged**